analyses/tmb-vaf-longitudinal/02-create-corplots.Rmd

```diff
@@ -0,0 +1,112 @@
+scratch_dir <- file.path(root_dir, "scratch")
+analysis_dir <- file.path(root_dir, "analyses", "tmb-vaf-longitudinal") 
+input_dir <- file.path(analysis_dir, "input")
+
+# Input files
+maf_autopsy_file <- file.path(scratch_dir, "maf_autopsy.tsv")
+oncoprint_goi_file <- file.path(input_dir, "oncoprint-goi-lists-OpenPedCan-gencode-v39.csv")
+timepoint_color_palette_file <- file.path(root_dir, "figures", "palettes", "timepoint_color_palette.tsv")
+
+# File path to plots directory
+plots_dir <-
+  file.path(analysis_dir, "plots")
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
+
+source(paste0(analysis_dir, "/util/function-create-corplot.R"))
+source(paste0(root_dir, "/figures/theme.R"))
+```
+
+## Read in data and process
+```{r read-input-files}
+# Read maf_autopsy file generated from step `01-process-data.Rmd`
+maf_autopsy <- readr::read_tsv(maf_autopsy_file, guess_max = 100000, show_col_types = FALSE)
+
+# Read oncoprint_goi_file
+# We will use the following list of genes and then we will create a label to select the ones for the corplots.
+oncoprint_goi <- read.csv(oncoprint_goi_file, stringsAsFactor = FALSE) %>%
+  select(LGAT, Embryonal.tumor, HGAT, Other) 
+
+# let's create a list with the genes to be used for the corplots
+oncoprint_goi <- data.frame(goi_list = unlist(oncoprint_goi)) 
+
+# remove empty rows
+oncoprint_goi <- oncoprint_goi[!apply(oncoprint_goi == "", 1, all), ] 
+oncoprint_goi <- data.frame(oncoprint_goi)
+
+# Read timepoint_color_palette file
+timepoint_color_palette <- readr::read_tsv(timepoint_color_palette_file, guess_max = 100000, show_col_types = FALSE)
+```
+
+## Corplots for each Patient case and per biospecimen_id and timepoint
+
+```{r create-corplot-please-wait}
+# We will define samples based on the "Kids_First_Participant_ID" column 
+samples <- unique(as.character(maf_autopsy$Kids_First_Participant_ID))
+print(samples)
+
+for (i in seq_along(samples)) {
+  print(i)
+  maf_sub <- maf_autopsy %>%
+    filter(Kids_First_Participant_ID == samples[i])
+  timepoints_other_plots <- unique(maf_sub$timepoints_other)
+  timepoints_other_plots <- timepoints_other_plots[!timepoints_other_plots == "4"]
+  print(timepoints_other_plots)
+  timepoints_deceased_plots <- unique(maf_sub$timepoints_deceased)
+  timepoints_deceased_plots <- timepoints_deceased_plots[!timepoints_deceased_plots %in% c("1", "2", "3")]
+  print(timepoints_deceased_plots)
+  
+# Run corplot
+  for (t in seq_along(timepoints_deceased_plots)){
+    for (k in seq_along(timepoints_other_plots)){
+      fname <- paste0(plots_dir, "/", samples[i], "-", timepoints_other_plots[k], "-vs-", timepoints_deceased_plots[t], "-vaf-corplot.pdf")
+      print(fname)
+      p <- create_corplot(maf = maf_sub,
+                          timepoints_other_plot = timepoints_other_plots[k],
+                          timepoints_deceased_plot = timepoints_deceased_plots[t],
+                          sid = samples[i])
```

Comment:
These plots look great! I think though some of the labels are not being printed bc there is too much overlap as indicated by the following warnings: 

`
Warning: ggrepel: 84 unlabeled data points (too many overlaps). Consider increasing max.overlaps
`

I would suggest tweaking the plotting function (in the util folder) to either make the text smaller, etc.

Reply:
I don't think this is an actual issue as `ggrepel` places the labels in a way they don't overlap with each other. Here we use the `geom_text_repel `function for this. I think that it would be problematic only if other warnings were present. For more information see here: [https://ggrepel.slowkow.com/articles/examples.html#overview](https://ggrepel.slowkow.com/articles/examples.html#overview).

Another way to go around this could have been to set `max.overlaps=Inf`. Should we do that? @jharenza 

But while thinking about this problem, I realized that we should add `set.seed` to ensure that the same data points are plotted every time, see a6fc0efa6c502be8afd3efa89ca5d766e397a4dd
So, thank you for that! :)